### PR TITLE
Add a PerfKitExplorer dashboard for every benchmark.

### PR DESCRIPTION
**Pull request by yuyantingzero**

### CODE REVIEW

**Comment by cmccoy** [comment]
Would you mind:

* Pretty-printing these ([`jq`](http://stedolan.github.io/jq/) does a fine job, as does Python's `json.dumps(..., indent=2)`).
* Naming each JSON file after the benchmark it tracks.

**Comment by yuyantingzero** [comment]
PTAL

File: tools/dashboard/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+This folder contains a list of dashboards for every benchmark in
+PerfKitBenchmarker. The dashboards can be uploaded and used by PerfKitExplorer.
+
+See:
+
+`https://github.com/GoogleCloudPlatform/PerfKitExplorer` for How to setup
+PerfKitExplorer.
+
+## Setup dashboards
```
Comment (voellm) [inline tools/dashboard/README.md]: I dont see how this describes what the user needs to do to deploy the dashboards to the explorer.   The sed commands wont really do anything as-is.

**Comment by voellm** [comment]
In addition to the readme.md in the dashboards directory I think you need to have help in the top level readme.md as well.

Also I think the dashboards should be rooted at the top level vs under tools.

**Comment by cmccoy** [comment]
I like them under `/tools` - keeping the top-level directory uncluttered is nice.

File: tools/dashboard/README.md (hunk 1 above)
Comment (yuyantingzero) [inline tools/dashboard/README.md]: Added more details in root README and dashboards/README

**Comment by yuyantingzero** [comment]
Moved to root directory and renamed the folder to dashboards. Since the next step will be adding integration tests for benchmarks. This will be more than a tool.

File: README.md [hunk 2]
```diff
@@ -377,6 +377,24 @@ file we have added a lot of comments into the code to make it easy to;
...
+
+To use pre-built dashboards
+* See PerfKitExplorer [README.md]
+(https://github.com/GoogleCloudPlatform/PerfKitExplorer/blob/master/README.md)
+for how to setup PerfKitExplorer and how to setup Google BigQuery dataset and table. 
+* Upload PerfKitBenchmarker results (/tmp/perfkitbenchmarker/run_*/perfkitbenchmarker_results.json) into BigQuery by running the following command (if you upload to a new BigQuery Table, a [schema]
+(https://github.com/GoogleCloudPlatform/PerfKitExplorer/blob/master/data/samples_mart/results_table_schema.json)
+file is needed):
+
+        bq load --source_format=NEWLINE_DELIMITED_JSON BIGQUERY_PROJECT_ID:BIGQUERY_DATASET.BIGQUERY_TABLE RESULTS_FILE (BIGQUERY_SCHEMA)
```
Comment (cmccoy) [inline README.md]: Any reason for preferring this over the `pkb` flags `--bigquery_table` / `--bq_project`?

File: dashboards/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# README
+
+This folder contains a list of pre-built dashboards for every benchmark in PerfKitBenchmarker and a summary dashboard for all benchmarks. The dashboards can be used by PerfKitExplorer.
+
+See:
+
+`https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/README.md` for setting up PerfKitExplorer and load data into Google BigQuery.
```
Comment (cmccoy) [inline dashboards/README.md]: Nit: use a markdown link here.

File: README.md (hunk 2 above)
Comment (yuyantingzero) [inline README.md]: The flags only works if the table already exists. Added the flag option and mentioned that it only works if the table exists.

File: dashboards/README.md (hunk 3 above)
Comment (yuyantingzero) [inline dashboards/README.md]: Done.

**Comment by yuyantingzero** [comment]
PTAL

**Comment by voellm** [comment]
By the way I am OK with this under tools but my vote is at the top layer.

The reason I don't prefer tools is that these are not a tool... the are a
feature set.

File: README.md [hunk 4]
```diff
@@ -377,6 +377,29 @@ file we have added a lot of comments into the code to make it easy to;
 
 Even with lots of comments we make to support more detailed documention.  You will find the documatation we have on the [Wiki pages] (https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/wiki).  Missing documentation you want?  Start a page and/or open an [issue] (https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues) to get it added.
 
+
+HOW TO VISUALIZE PerfKitBenchmarker RESULTS
```
Comment (cmccoy) [inline README.md]: PerfKitExplorer isn't the *only* way to visualize the results. Maybe make this "Visualizing the results in PerfKitExplorer"?

File: README.md [hunk 5]
```diff
@@ -377,6 +377,29 @@ file we have added a lot of comments into the code to make it easy to;
...
+==================
+The dashboards directory contains a list of pre-built dashboards for every benchmark in PerfKitBenchmarker and a summary dashboard for all benchmarks. The dashboards can be used by PerfKitExplorer.
```
Comment (cmccoy) [inline README.md]: I would recommend keeping this section very short - maybe just this sentence and your "see  `dashboards/README.md`" link.

Comment (cmccoy) [inline README.md]: (and moving anything only covered here to `dashboards/README.md`).

File: dashboards/README.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+for setting up PerfKitExplorer and load data into Google BigQuery.
+
+## Setup dashboards
+* Once the PerfKitExplorer is deployed, config the the dashboards with your BIGQUERY_PROJECT_ID, BIGQUERY_DATASET and BIGQUERY_TABLE used by PerfKitExplorer by running the following command in this directory:
+
+        sed -i  's/{{ project_id }}/BIGQUERY_PROJECT_ID/g' *.json
+        sed -i  's/{{ dataset_name }}/BIGQUERY_DATASET/g' *.json
+        sed -i  's/{{ table_name }}/BIGQUERY_TABLE/g' *.json
+
+* Go to PerfKitExplorer site deployed by AppEngine: http://version-dot-EXPLORER_PROJECT_ID.appspot.com
```
Comment (cmccoy) [inline dashboards/README.md]: Grammar: "Go to the" and deployed by -> deployed on.

File: dashboards/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+* Click "Upload" button in the "PerfKit Dashboard Administration" to upload the dashboards.
```
Comment (cmccoy) [inline dashboards/README.md]: Could you clarify this? (in the "PerfKit Dashboard Administration"...menu? Section?)

File: dashboards/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# README
+
+This folder contains a list of pre-built dashboards for every benchmark in PerfKitBenchmarker and a summary dashboard for all benchmarks. The dashboards can be used by PerfKitExplorer.
```
Comment (cmccoy) [inline dashboards/README.md]: remove "a list" - you have teh dashboards themselves.
Link to perfkitexplorer here.

    [PerfKitExplorer](http://github.com/GoogleCloudPlatform/PerfKitExplorer)

File: dashboards/README.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+See PerfKitBenchmarker [README.md]
+(https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/README.md)
+for setting up PerfKitExplorer and load data into Google BigQuery.
+
+## Setup dashboards
+* Once the PerfKitExplorer is deployed, config the the dashboards with your BIGQUERY_PROJECT_ID, BIGQUERY_DATASET and BIGQUERY_TABLE used by PerfKitExplorer by running the following command in this directory:
```
Comment (cmccoy) [inline dashboards/README.md]: the PerfKitExplorer -> PerfKitExplorer
config -> configure

**Comment by voellm** [comment]
This CL is going to move to the PerfKitExplorer project.

**Pull request closed without merging**